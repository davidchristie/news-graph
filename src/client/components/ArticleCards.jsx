import React, { PropTypes } from 'react'
import ArticleCard from './ArticleCard'

export default class ArticleCards extends React.Component {
  render () {
    const articles = this.props.articles
    return (
      <div className='card-columns'>
        {articles.map(article =>
          <ArticleCard
            key={article.id}
            article={article}
          />
        )}
      </div>
    )
  }
}

ArticleCards.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
}
