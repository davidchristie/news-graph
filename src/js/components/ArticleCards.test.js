/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import ArticleCards from './ArticleCards'

const exampleArticles = [
  {
    id: 1,
    url: 'http://www.nature.com/news/ancient-retroviruses-emerged-half-a-billion-years-ago-1.21274'
  },
  {
    id: 2,
    url: 'https://www.theguardian.com/world/2017/jan/11/chamber-of-rats-mexico-parliaments-name-changed-in-google-maps-prank'
  },
  {
    id: 3,
    url: 'https://vimeo.com/192574852'
  },
  {
    id: 4,
    url: 'http://www.aljazeera.com/indepth/features/2017/01/india-love-hand-books-170108073151216.html'
  }
]

test('ArticleCards renders correctly', () => {
  const tree = renderer.create(
    <ArticleCards articles={exampleArticles} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
