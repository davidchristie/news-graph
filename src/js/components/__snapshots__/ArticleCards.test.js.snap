exports[`test ArticleCards renders correctly 1`] = `
<div
  className="album text-muted">
  <div
    className="container">
    <div
      className="row">
      <div
        className="card-group">
        <div
          className="card col-md-6 col-lg-4">
          <a
            className="embedly-card"
            href="http://www.nature.com/news/ancient-retroviruses-emerged-half-a-billion-years-ago-1.21274" />
          <p>
            Some details about the article.
          </p>
          <a
            className="btn"
            onClick={[Function]}
            style={Object {}}>
            Details
          </a>
        </div>
        <div
          className="card col-md-6 col-lg-4">
          <a
            className="embedly-card"
            href="https://www.theguardian.com/world/2017/jan/11/chamber-of-rats-mexico-parliaments-name-changed-in-google-maps-prank" />
          <p>
            Some details about the article.
          </p>
          <a
            className="btn"
            onClick={[Function]}
            style={Object {}}>
            Details
          </a>
        </div>
        <div
          className="card col-md-6 col-lg-4">
          <a
            className="embedly-card"
            href="https://vimeo.com/192574852" />
          <p>
            Some details about the article.
          </p>
          <a
            className="btn"
            onClick={[Function]}
            style={Object {}}>
            Details
          </a>
        </div>
        <div
          className="card col-md-6 col-lg-4">
          <a
            className="embedly-card"
            href="http://www.aljazeera.com/indepth/features/2017/01/india-love-hand-books-170108073151216.html" />
          <p>
            Some details about the article.
          </p>
          <a
            className="btn"
            onClick={[Function]}
            style={Object {}}>
            Details
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
